Magda

mabuszka opened the pull request:
the last version of our project

Wicix committed:
Copied code from winter semester

asimon7 committed:
test

test

mabuszka committed:
Update README.md

test

mabuszka committed:
Update README.md

test

mabuszka committed:
adds basic package files and one function

mabuszka committed:
adds basic package files and one function

asimon7 committed:
Merge branch 'magda' into artur

asimon7 committed:
adding author

asimon7 committed:
add author

asimon7 committed:
Merge remote-tracking branch 'origin/artur' into artur

asimon7 committed:
edit global

make_playlist_df and str_his_with_playlists

mabuszka committed:
adds functions that use data.table

Adds functions that use data table and updates description and man

mabuszka committed:
deletes files from last semester from this branch

mabuszka committed:
modifies decription

asimon7 committed:
add functions/ fix notes

str_his_with_playlists, prepare_playlists

asimon7 committed:
Merge remote-tracking branch 'origin/artur' into artur

mabuszka committed:
adds and edits streaming history functions

Added :
-a new function combining number_of_tracks_hour/weekday into one function, but without period filtering within
- a function for period filtering streaming history
- helper function for capitalizing words
Also some minor changes in rest of the functions.

asimon7 committed:
edit functions

preapare_playlists, str_hist_with_playlists

mabuszka committed:
add .travis.yml

mabuszka committed:
Update .gitignore

mabuszka committed:
adds travis

mabuszka committed:
Merge remote-tracking branch 'origin/magda' into magda

mabuszka committed:
fixes adding travis

mabuszka committed:
Merge remote-tracking branch 'origin/artur' into magda

Wicix committed:
add reading functions

Wicix committed:
add complete_streaming_history

asimon7 committed:
edit function

str_his_with_playlists

mabuszka committed:
adds a function

Wicix committed:
Correct prepare_streaming_history

mabuszka committed:
Auto stash before merge of "magda" and "origin/artur"

mabuszka committed:
Merge remote-tracking branch 'origin/artur' into magda

asimon7 committed:
edit function

str_hist_with_playlists into two different

mabuszka committed:
Merge remote-tracking branch 'origin/wiktor' into magda

mabuszka committed:
add documentation

asimon7 committed:
Merge remote-tracking branch 'origin/magda' into artur

asimon7 committed:
Auto stash before merge of "artur" and "origin/magda"

Wicix committed:
Correct make_search_queries_dt

mabuszka committed:
edits functions

edits filter_search_queries and make_search_queries_dt functions

mabuszka committed:
edits documentation and functions

asimon7 committed:
add function

continuous_listening

mabuszka committed:
edits description

edits travis badge

mabuszka committed:
edits readme

asimon7 committed:
add function

compare_history

mabuszka committed:
Merge remote-tracking branch 'origin/artur' into magda

mabuszka committed:
edit str_hist_with_playlists_wide, add plot_in_playlists_count

added a column "in_any" - TRUE if track is in any of the playlists,
added plot_in_playlists_count function

mabuszka committed:
adds plot_playlist_popularity, edits documentation

adds plot_playlist_popularity, 
edits documentation and adds NULL assignments to eliminate check notes

mabuszka committed:
modify filter_streaming_history

modify filter_streaming_history date filtering so that the intervals work like [start_date, end_date] as opposed to earlier [start_date, end_date)

mabuszka committed:
adds make_two_users_dt, plot_daily_comparison functions, fixes spelling

adds two functions to compare two users  daily play time, 
corrects spelling in description and documentation

mabuszka committed:
changes the name of function

change function name to 'plot_day_comparison_

asimon7 committed:
add function

sessions_visualise
sessions_length
longest_session

asimon7 committed:
Merge remote-tracking branch 'origin/magda' into artur

asimon7 committed:
edit/ add function, fix notes

rewrite continuous_listening
take computation from sessions_visualize and put them into sessions_intervals

mabuszka committed:
Merge remote-tracking branch 'origin/artur' into magda

asimon7 committed:
edit function/ create app

edit continuous_listening,
create app in www

mabuszka committed:
adds common theme and colors to plots

- adds common theme, specified in theme_spotifyviz, to all plots, changes colors 
- adds theme_spotifyviz function

mabuszka committed:
Merge remote-tracking branch 'origin/artur' into magda

mabuszka committed:
fixes minor bugs

fixes labels on axis and position of the legend

asimon7 committed:
shiny

add sidebar

mabuszka committed:
Merge remote-tracking branch 'origin/artur' into magda

mabuszka committed:
edit ui

asimon7 committed:
Merge remote-tracking branch 'origin/magda' into artur

asimon7 committed:
Auto stash before merge of "artur" and "origin/magda"

mabuszka committed:
modify and add functions

adds from_sec_to_min_sec - a helper function to translate number of seconds into  X min Y s format
modifies most_played, most_skipped, longest_session so that they are able to output data tables in a form convenient to read (eg. spaces instead of "_" i column names)

mabuszka committed:
edits from_sec_to_min_sec, longest_session

edit a function to behave correctly when either mins or secs is equal to 0,
corrects an unnecessary commenting of a line in longest_session

mabuszka committed:
change from_sec_to_min_sec, add make_session_stats, session_for_view

-change forom_sec_to_min_sec into from_sec_to_hms - changes fromat to Xh Ymin Zs
- add make-session stats function to make a summary table with session stats
- add session_for_view - option for viewing session from longest_session made into a separate function

mabuszka committed:
fixes a bug in make_session_stats

asimon7 committed:
Merge remote-tracking branch 'origin/magda' into artur

mabuszka committed:
make eventReactive file inputs

asimon7 committed:
Merge remote-tracking branch 'origin/magda' into artur

asimon7 committed:
Auto stash before merge of "artur" and "origin/magda"

mabuszka committed:
add tables tab content and modify input tab content

asimon7 committed:
edit app

add plots  search queries

asimon7 committed:
Merge remote-tracking branch 'origin/magda' into artur

mabuszka committed:
add reactive ui for generating default values for date input

Wicix committed:
Merge remote-tracking branch 'origin/magda' into wiktor

mabuszka committed:
Merge remote-tracking branch 'origin/artur' into magda

asimon7 committed:
edit app

add plot in plot streaming_history

asimon7 committed:
Merge remote-tracking branch 'origin/magda' into artur

mabuszka committed:
Merge remote-tracking branch 'origin/artur' into magda

mabuszka committed:
fixed labeling mistakable

asimon7 committed:
Merge remote-tracking branch 'origin/magda' into artur

mabuszka committed:
fix bug with as_percentage buttons

asimon7 committed:
track and session plots in streaming history

mabuszka committed:
add new summary function, add it to app

add new summary function,
add it to app, 
make minor changes in app

mabuszka committed:
Merge remote-tracking branch 'origin/artur' into magda

asimon7 committed:
edit session plot

add default mins

mabuszka committed:
edit plot_track_count, _by_period, theme, and from_sec_to_hms

mabuszka committed:
Merge remote-tracking branch 'origin/artur' into magda

Wicix committed:
Merge remote-tracking branch 'origin/magda' into wiktor

Wicix committed:
Add test for function make_streaming_history_dt and add test data

mabuszka committed:
edits functions, edits app

edits app,
edit from_sec_to_hms - to work on integer vectors
edit make session stat for better table

Wicix committed:
add test for reading functions and correct typo

Correct typo for make_search_queries_dt.R
add test function and test files for reading functions

mabuszka committed:
add compare table to app

Wicix committed:
Add file for testing and test for preparing functions tbc.

These two functions are prepare_streaming_history() and complete_streaming_history()

mabuszka committed:
adds raective time input and fixes bug in plot_track_count_by_period

asimon7 committed:
Merge remote-tracking branch 'origin/magda' into artur

asimon7 committed:
add track for package/fix notes

asimon7 committed:
add files

asimon7 committed:
create vignette

Wicix committed:
commit before mergin from arthur, add and replace test files

Wicix committed:
Merge remote-tracking branch 'origin/magda' into wiktor

Wicix committed:
Merge remote-tracking branch 'origin/artur' into wiktor

mabuszka committed:
add test vignnete

asimon7 committed:
Merge remote-tracking branch 'origin/magda' into artur

Wicix committed:
Add functions and files for test, and repair reading functions

asimon7 committed:
edit vignette

asimon7 committed:
add files needed to vignette

Wicix committed:
Add next test files

mabuszka committed:
Merge remote-tracking branch 'origin/wiktor' into magda

mabuszka committed:
Merge remote-tracking branch 'origin/artur' into magda

Wicix committed:
Repair reading functions and add "testthat" to suggests

Wicix committed:
Add .Rda data files to data folder

mabuszka committed:
change vignette

mabuszka committed:
Merge remote-tracking branch 'origin/wiktor' into magda

mabuszka committed:
add vignette

Wicix committed:
Add whole night effect

Improvement of tests, save R files in Data folder, spelling correction, add me as author of project

Wicix committed:
Merge remote-tracking branch 'origin/magda' into wiktor

Wicix committed:
Delete www folder and prepares for adding to master

asimon7 committed:
Merge remote-tracking branch 'origin/magda' into artur

asimon7 committed:
Merge remote-tracking branch 'origin/wiktor' into artur

asimon7 committed:
Auto stash before merge of "artur" and "origin/wiktor"

asimon7 committed:
edit viegnette

fix title conflict

asimon7 committed:
eddit vignette

fix title conflict

asimon7 committed:
edit global

asimon7 committed:
update readme

add new link to shiny app

mabuszka committed:
update readme

asimon7 committed:
edit app

fix search queries for user 2

mstaniak commented on DESCRIPTION:
```diff
@@ -0,0 +1,37 @@
+Package: spotifyviz
+Title: Visualize Spotify Data
+Version: 0.0.0.9000
+Authors@R: 
```
jeżeli uważacie, że miałem odpowiedni wpływ na projekt, możecie mnie dodać do autorów,
Depends zwykle daje się nad Imports

mstaniak replied:
po namyśle co do autorstwa: napiszcie tylko w apce, że konsultowaliście ze mną / pomogłem

mstaniak commented on NAMESPACE:
```diff
@@ -0,0 +1,44 @@
+import(data.table)
+import(ggplot2)
+import(jsonlite)
+import(lubridate)
```
dużo funkcji używacie z lubridate? Możecie lepiej mieć dla nich importFrom() - ale to mało istotne

mstaniak commented on R/capitalize.R:
```diff
@@ -0,0 +1,12 @@
+#' 
+#' Capitalize first letter in the string.
+#' 
+#' @param string A string that you want to capitalize. 
```
ja lubię robić mniej enterów, a za to mieć koniec pusty #' na koniec dokumentacji - jak uważacie

mstaniak replied:
raczej można mieć jeden plik na takie małe funkcje w stylu utility_functions, auxiliary_function etc

mstaniak commented on R/clean_playlist.R:
```diff
@@ -0,0 +1,13 @@
+#' A dataset, selected from Wiktor's data from spotify,result of using prepare_playlists()
+#'on JSON file, prepared for testing and vignettes
+#
+#' @format A data table with 159 rows and 3 variables:
```
data.table

mstaniak commented on R/clean_search_queries.R:
```diff
@@ -0,0 +1,13 @@
+#' A dataset with search queries,result of using make_search_queries_dt() on JSON file, 
+#' selected from Wiktor's data from spotify, prepared for testing and vignettes.
+#
+#' @format A data table with 10 rows and 3 variables:
```
data.table

mstaniak commented on R/clean_stream_his.R:
```diff
@@ -0,0 +1,9 @@
+#'selected from Wiktor's data from spotify,
+#' prepared for testing and vignettes.
+#
+#' @format A data table with 100 rows and 7 variables:
```
data.table

mstaniak commented on R/clean_playlist.R:
```diff
@@ -0,0 +1,13 @@
+#' Clean playlist
```
mozna dać wszystkie dane do jednego pliku i opisać je też w jednym pliku pomocy

mstaniak commented on R/compare_history.R:
```diff
@@ -0,0 +1,35 @@
+#'
+#' Shows the same artists/ artists and tracks from history of two different users.
+#'
+#' @param streaming_history_1 A data.table containing streaming history, 
```
data.table processed by the `prepare_streaming_history` function lub coś podobnego

mstaniak commented on R/complete_streaming_history.R:
```diff
@@ -0,0 +1,18 @@
+#' @return A data table containing streaming history from spotify suited for being used 
+#' in rest of the functions from the package.
+#'
+#' @export
```
może przykłady z \donotrun?

mstaniak commented on R/continuous_listening.R:
```diff
@@ -0,0 +1,36 @@
+#' Measure the number of continuous listening sessions.
+#' 
+#' Listening number informs about number of songs listened in a row.
```
niejasny opis (coś nie gra gramatycznie albo jest niedopowiedziane)

mstaniak commented on R/filter_search_queries.R:
```diff
@@ -0,0 +1,23 @@
+#' @seealso \code{\link{filter_streaming_history}}
+
+
+filter_search_queries <- function(search_queries, start_date, end_date) {
```
funkcja jest mniej ogólna niż wskazuje nazwa - sortuje tylko po dacie

mstaniak commented on R/filter_streaming_history.R:
```diff
@@ -0,0 +1,23 @@
+#' @seealso \code{\link{filter_search_queries}}
+
+
+filter_streaming_history <- function(streaming_history, start_date, end_date) {
```
funkcja jest mniej ogólna niż wskazuje nazwa - filtruje tylko po dacie

mstaniak commented on R/from_sec_to_hms.R:
```diff
@@ -0,0 +1,34 @@
+
+from_sec_to_hms <- function(x) {
+  
+  single <- function(x) {
```
nie wiem, czy ta funkcja jest często wywoływana, ale single lepiej mieć poza nią, żeby jej nie tworzyć wiele razy

mstaniak replied:
a za to móc testować

mstaniak commented on R/how_long_listened.R:
```diff
@@ -0,0 +1,42 @@
+#' @import data.table
+#' @import lubridate
+
+how_long_listened <- function(streaming_history, start_date, end_date, as_percentage = FALSE, 
```
nazwa niespójna z wcześniejszymi nazwami (nie zaczyna się od czasownika)

mstaniak commented on R/how_long_listened.R:
```diff
@@ -0,0 +1,42 @@
+  start_time <- s_played <- NULL
+  end_date <- as_date(end_date)
+  start_date <- as_date(start_date)
+  suma <- round(streaming_history[start_time >= start_date & start_time <= end_date, sum(s_played)], 0)
```
nazwa niespójna z innymi - po polsku

mstaniak commented on R/how_many_skipped.R:
```diff
@@ -0,0 +1,39 @@
+#' @import data.table
+
+
+how_many_skipped <-
```
po co ten enter?

mstaniak commented on R/longest_session.R:
```diff
@@ -0,0 +1,28 @@
+#' @export
+#'
+#' @import data.table
+
```
po co te entery?

mstaniak commented on R/make_search_queries_dt.R:
```diff
@@ -0,0 +1,25 @@
+    file_path <- paste(folder_path, file_path, sep = "/")
+  }
+  read_file <- data.table(jsonlite::fromJSON(file_path))
+  read_file[, list(date = ymd(date), platform, country)]
```
niespójne z wcześniejszym kodem (list zamiast .)

mstaniak commented on R/make_session_stats.R:
```diff
@@ -0,0 +1,31 @@
+#' Must have columns "artist_name", "track_name", "s_played", "start_time".
+#'
+#' @export
+#'
```
@importFrom data.table data.table

mstaniak commented on R/make_session_stats.R:
```diff
@@ -0,0 +1,31 @@
+make_session_stats <- function(session_dt) {
+  start_time <- end_time <- . <- s_played <- NULL
+  table <- data.table(
+    `Session start` = c(substr(
```
po co te c()?

mstaniak commented on R/make_streaming_history_dt.R:
```diff
@@ -0,0 +1,21 @@
+#' Make data table with streaming history
```
to można włożyć do jednego pliku z innymi funkcjami make_

mstaniak commented on R/make_summary_dt.R:
```diff
@@ -0,0 +1,44 @@
+    hm_different_artists <- uniqueN(filtered[, .(artist_name)])
+    hl_listened <- how_long_listened(streaming_history, start_date, end_date, as_percentage)
+    
+    summary <- data.table(
```
data.table::data.table

mstaniak commented on R/number_of_skipped_played_tracks.R:
```diff
@@ -0,0 +1,76 @@
+      streaming_history <- streaming_history[(skipped),]
+      y_lab = "Number of tracks skipped"
+    }
+    else{
```
styl

mstaniak commented on R/number_of_skipped_played_tracks.R:
```diff
@@ -0,0 +1,76 @@
+                            end_time = end_time)]
+      y_lab = "Number of tracks played"
+    }
+    viz <-
```
po co ten enter

mstaniak commented on R/plot_in_playlists_count.R:
```diff
@@ -0,0 +1,90 @@
+      theme_spotifyviz()
+  }
+  
+  else{
```
styl

mstaniak commented on R/plot_playlist_popularity.R:
```diff
@@ -0,0 +1,67 @@
+      
+    }
+    
+    else{
```
styl

mstaniak commented on R/plot_track_count.R:
```diff
@@ -0,0 +1,72 @@
+#' @import lubridate
+
+
+plot_track_count <-
```
zbędny enter

mstaniak commented on README.md:
```diff
@@ -1 +1,32 @@
+An R tool and Shiny application to explore music tastes based on Spotify data.
+The application is available 12.06.2020.
+
+https://asimon.shinyapps.io/SpotifyViz/
```
zamienić na markdownowy link, usunąć datę

mstaniak commented on README.md:
```diff
@@ -1 +1,32 @@
+    - popularity of different playlists
+    - visualizing hourly/daily lstening habits
+    - the comparison of listesting habits on one day for two uses
+    
```
dodajcie, proszę, info o tym, że napisane w ramach przedmiotu + iink do repo przedmiotu

mstaniak commented on tests/spelling.R:
```diff
@@ -0,0 +1,3 @@
+if (requireNamespace("spelling", quietly = TRUE))
```
ładnie

mstaniak commented on tests/testthat/test-cleaning_functions.R:
```diff
@@ -0,0 +1,28 @@
+context("Cleaning functions")
```
context jest chyba deprecated

mstaniak commented on tests/testthat/test-reading_functions.R:
```diff
@@ -0,0 +1,30 @@
+context("Reading JSON Files")
```
ładnie

mstaniak commented on vignettes/basic.Rmd:
```diff
@@ -0,0 +1,352 @@
+
+---
```
zasadniczo winietka będzie potrzebowała zmian, ale na razie zostawcie, i tak jest dobra

mstaniak commented on R/most_played.R:
```diff
@@ -0,0 +1,36 @@
+  }
+  else {
+    most_played_dt <- streaming_history[skipped == FALSE, .N,
+                                        by = eval(paste(track_or_artist, "_name", sep = ""))][order(-N)][order(-N)][1:how_many, ]
```
w ogólności 1:how_many jest niebezpieczne (np how_many = 0), więc wymagałoby asserta, w kontekście apki jest OK

mstaniak reviewed (changes requested):
Ogólnie 
  
- styl kodu mógłby być trochę lepszy (brakujące / nadmiarowe spacje / entery)
- angielski mógłby być troszkę lepszy
- chciałbym jeszcze zobaczyć kod apki, a tu go nie widzę - gdzie jest?

asimon7 committed:
edit app

Wicix committed:
Merge remote-tracking branch 'origin/magda' into wiktor

Wicix committed:
Merge remote-tracking branch 'origin/app' into wiktor

Copy App

Wicix committed:
Updates make_streaming_history 

Now you can use multiply data sets from same user

mabuszka committed:
Merge remote-tracking branch 'origin/wiktor' into magda

mabuszka committed:
Change text size in theme, add utility functions to one file

Change text size in theme,
add utility functions to one file,
deletes some  unnecessary enters

mabuszka committed:
deletes apps code, corrects mistake, corrects data documantection

asimon7 committed:
edit description

continuous listening

asimon7 committed:
Merge remote-tracking branch 'origin/magda' into artur

asimon7 committed:
edit description

continuous listening

mabuszka committed:
renames functions, deletes unnecessary enters

renames how_long_listened to measure_how_long_listened
renames how_many_skipped to count_skipped

mabuszka committed:
corrects uses of renamed functions

mabuszka committed:
Merge remote-tracking branch 'origin/artur' into magda

mabuszka committed:
edits vignette

mabuszka committed:
update readme

mabuszka committed:
corrects date filtering mistakes
